### DISCUSSION TITLE
Ploting is much slower on Linux than on Windows

### DISCUSSION THREAD

**Discussion by mjchow** [Chia-Network/chia-blockchain#6442] [Support]
I'm using the same hardware configuration and the same parameters on both Linux and Windows

-b 6000 -r 8

the CPU utilization on Windows is 400%
CPU utilization on Linux can't even reach 200%

**Comment by heus12** [comment]
no，linux is 10%  faster than windows

if you compile and optimization linux kernel, that can 20%-40% faster  than windows

**Reply by mjchow** [reply]
Which linux distribution do you use?

**Reply by heus12** [reply]
> Which linux distribution do you use?

 Every linux distribution can use, but you need to compile and optimize the Linux kernel yourself

**Reply by mjchow** [reply]
F1 complete, time: 659.182 seconds. CPU (56.44%) this is in ubuntu20.04
F1 complete, time: 200.342 seconds. CPU (189.32%) this is in windows 10

Why is CPU utilization so different for the same plot parameter ？ my plot parameter is (-k 32 -b 8000 -u 128 -r 8)

**Comment by loppefaaret** [comment]
Windows wil by default ensure that any SSD based media gets TRIM.
On the other hand, under most Linux distro's, you need to specify this yourself, with adding the mount option `discard` to an SSD based media.
Perhaps this is an issue you are facing ?

**Reply by mjchow** [reply]
i am not sure. 
F1 complete, time: 659.182 seconds. CPU (56.44%)     this is in ubuntu20.04 
F1 complete, time: 200.342 seconds. CPU (189.32%)   this is in windows 10

Why is CPU utilization so different for the same plot parameter ？  my plot parameter is (-k 32 -b 8000 -u 128 -r 8)

**Reply by loppefaaret** [reply]
Cause the drive is being managed differently. What does `mount | grep discard` return ? it should return any SSD used for chia plotting

**Reply by mjchow** [reply]
`mount | grep discard` There is no return.

But in terms of completion time, windows 10 is much faster than Ubuntu 20.04 in the same phase

**Reply by loppefaaret** [reply]
if you get the drives mounted with `discard` mount option(as long as they are SSD), ubuntu would be faster than windows!

**Reply by loppefaaret** [reply]
chia decentral youtube channel has gone through this extensively

**Reply by mjchow** [reply]
> Cause the drive is being managed differently. What does `mount | grep discard` return ? it should return any SSD used for chia plotting



I mount two nvme SSDs to / media / t0 and / media / T1

nvme2n1     259:0    0 931.5G  0 disk 
├─nvme2n1p1 259:2    0    16M  0 part 
└─nvme2n1p2 259:3    0 931.5G  0 part /media/T0
nvme1n1     259:1    0 931.5G  0 disk 
├─nvme1n1p1 259:4    0    16M  0 part 
└─nvme1n1p2 259:5    0 931.5G  0 part /media/T1

**Reply by antonello22** [reply]
i have a i5 cpu on ubuntu 20 and my F1 time is 123 seconds with my ssd to pcie card.... 
just be sure you are using your ssd as a directory

**Reply by loppefaaret** [reply]
`sudo mount -t XXXX -o discard,noatime /dev/nvme1n1p2 /media/T1`
`sudo mount -t XXXX -o discard,noatime /dev/nvme2n1p2 /media/T0`

**Reply by mjchow** [reply]
sudo mount -t ntfs -o discard,noatime /dev/nvme1n1p2 /media/T1
sudo mount -t ntfs -o discard,noatime /dev/nvme2n1p2 /media/T0

but `mount | grep discard` There is still no return.

**Reply by mjchow** [reply]
> `sudo mount -t XXXX -o discard,noatime /dev/nvme1n1p2 /media/T1`
> `sudo mount -t XXXX -o discard,noatime /dev/nvme2n1p2 /media/T0`

I think the utilization rate of CPU multithreading in Linux is too low

**Reply by leonidcan** [reply]
@mjchow ntfs on Linux is slow. You should use native file system on *nix systems (ie ext4)

**Reply by mjchow** [reply]
> if you get the drives mounted with `discard` mount option(as long as they are SSD), ubuntu would be faster than windows!

My SSD is nvme protocol, and there is no trim, right？

**Reply by mjchow** [reply]
> @mjchow ntfs on Linux is slow. You should use native file system on *nix systems (ie ext4)

thank u.  @leonidcan 
After using ext4 in my SSD, the speed of plot is improved, but it is still slower than that under windows

**Reply by loppefaaret** [reply]
did you use discard with ext4 ?

**Reply by mjchow** [reply]
@loppefaaret 
I don't seem to be using discard
How to check if I used discard

i typed `mount | grep discard` is no return

**Reply by loppefaaret** [reply]
if there is no return, there is no drive mounted with discard, so mount them with discard, as above

**Reply by mjchow** [reply]
i mount with discard 
but there is no effect

**Comment by leonidcan** [comment]
I found that my linux is faster than windows (used on the same machine) but I am using software raid, how do I apply TRIM to RAID0 in mdadm ? Can't find good answer but I notice that plotting time go up overtime :(

**Reply by mjchow** [reply]
I feel that CPU utilization on Linux is lower than that on windows
But I don't know why
My CPU is e5-2695v3

**Comment by BrandtH22** [comment]
@mjchow , since this ticket was opened a number of years ago without further comments it seems this issue has been resolved. 
We will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions or comments.

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
